### PR TITLE
Implement

### PR DESCRIPTION
**Pull request by supl**
This PR implements the main part of the CLI using Scalar DB.

### CODE REVIEW

**Comment by supl** [comment]
@brfrn169 Thanks for your joining!
Please take a look when you are free :)

**Comment by supl** [comment]
@feeblefakie :) Thank you.

File: app/src/main/java/getstarted/Worklog.java [hunk 1]
```diff
@@ -0,0 +1,100 @@
...
+    tx.put(put2);
+
+    tx.commit();
+  }
+
+  public void today() throws Exception {
+    Key pKey =
+        new Key(new TextValue("date", DateTimeFormatter.BASIC_ISO_DATE.format(LocalDate.now())));
+    Scan scan = new Scan(pKey).forNamespace("plenty").forTable("worklog");
+    Scanner scanner = storage.scan(scan);
```
Comment (feeblefakie) [inline app/src/main/java/getstarted/Worklog.java]: For data that is updated through transactions, it should be scanned and read by transactions as well.
Scanning with storage would return unexpected (stale) values.

File: app/src/main/java/getstarted/Worklog.java [hunk 2]
```diff
@@ -0,0 +1,100 @@
...
+    Iterator<Result> iterator = scanner.iterator();
+
+    while (iterator.hasNext()) {
```
Comment (feeblefakie) [inline app/src/main/java/getstarted/Worklog.java]: It's usually thought to be better to use lambda in the current Java if possible.
We sometimes use `for` in the case where code is formatted too ugly by the google format, and we rarely use iterator explicitly.
```
  scanner.forEach(r -> {
      BigIntValue timestampValue = (BigIntValue) r.getValue("timestamp").get();
      TextValue logValue = (TextValue) r.getValue("log").get();
      Date d = new Date(timestampValue.get() * 1000);
      System.out.println(d.toString() + ": " + logValue.getString().get());
  });
```

File: app/src/main/java/getstarted/Worklog.java [hunk 3]
```diff
@@ -0,0 +1,100 @@
...
+      count = ((IntValue) result.get().getValue("count").get()).get();
+    }
+    count += 1;
+    Put put2 = new Put(pKey).withValue(new IntValue("count", count)).forNamespace("plenty")
+        .forTable("workcount");
+
+    tx.put(put1);
+    tx.put(put2);
+
+    tx.commit();
```
Comment (feeblefakie) [inline app/src/main/java/getstarted/Worklog.java]: We should catch `CommitConflictException` for the same reason.

File: app/src/main/java/getstarted/Worklog.java [hunk 4]
```diff
@@ -0,0 +1,100 @@
...
+    Optional<Result> result = tx.get(get);
+
+    if (result.isPresent()) {
+      count = ((IntValue) result.get().getValue("count").get()).get();
+    }
+    count += 1;
+    Put put2 = new Put(pKey).withValue(new IntValue("count", count)).forNamespace("plenty")
+        .forTable("workcount");
+
+    tx.put(put1);
```
Comment (feeblefakie) [inline app/src/main/java/getstarted/Worklog.java]: It's not fully documented yet, but we should catch CrudConflictExcecption for `put`, meaning the transaction was failed due to a conflicting transaction, and retry for better user experiences.

**Review by feeblefakie — COMMENTED** [review]
Sorry for the late review and thank you!
Left some comments and suggestions.

File: app/src/main/java/getstarted/Worklog.java [hunk 5]
```diff
@@ -0,0 +1,100 @@
...
+    tx.put(put2);
+
+    tx.commit();
+  }
```
Comment (brfrn169) [inline app/src/main/java/getstarted/Worklog.java:80]: You need to catch Exception and call `tx.abort()` for failure cases.

**Review by brfrn169 — COMMENTED** [review]
Thank you! I left a comment. Please take a look at it when you have time.

File: app/src/main/java/getstarted/Worklog.java [hunk 6]
```diff
@@ -0,0 +1,100 @@
...
+    Scan scan = new Scan(pKey).forNamespace("plenty").forTable("worklog");
+    Scanner scanner = storage.scan(scan);
+    Iterator<Result> iterator = scanner.iterator();
+
+    while (iterator.hasNext()) {
+      Result result = iterator.next();
+      BigIntValue timestampValue = (BigIntValue) result.getValue("timestamp").get();
+      TextValue logValue = (TextValue) result.getValue("log").get();
+      Date d = new Date(timestampValue.get() * 1000);
+      System.out.println(d.toString() + ": " + logValue.getString().get());
```
Comment (brfrn169) [inline app/src/main/java/getstarted/Worklog.java]: Nit: Unnecessary toString() call 
```suggestion
      System.out.println(d + ": " + logValue.getString().get());
```

**Comment by supl** [comment]
@feeblefakie 
@brfrn169 
Thanks for your reviews and feedbacks.
I revised the implementation according to your advises.
Please take a look again when you have free time.
Thank you :)

**Pull request closed without merging**